client-sdk/ts-web/rt/src/types.ts

```diff
@@ -800,3 +800,378 @@ export interface ContractsContractEvent {
+    offers: RoflmarketOffer[];
+    metadata: {[key: string]: string};
+}
+
+export interface RoflmarketProviderUpdate {
+    provider: Uint8Array;
+    nodes: PublicKey[];
+    scheduler_app: AppID;
+    payment_address: RoflmarketPaymentAddress;
+    metadata: {[key: string]: string};
+}
+
+export interface RoflmarketProviderUpdateOffers {
+    provider: Uint8Array;
+    add?: RoflmarketOffer[];
+    update?: RoflmarketOffer[];
+    remove?: OfferID[];
+}
+
+export interface RoflmarketProviderRemove {
+    provider: Uint8Array;
+}
+
+export interface RoflmarketInstanceCreate {
+    provider: Uint8Array;
+    offer: OfferID;
+    admin?: Uint8Array;
+    deployment?: RoflmarketDeployment;
+    term: RoflmarketTerm;
+    term_count: oasis.types.longnum;
+}
+
+export interface RoflmarketInstanceTopUp {
+    provider: Uint8Array;
+    id: MachineID;
+    term: RoflmarketTerm;
+    term_count: oasis.types.longnum;
+}
+
+export interface RoflmarketInstanceCancel {
+    provider: Uint8Array;
+    id: MachineID;
+}
+
+export interface RoflmarketInstanceExecuteCmds {
+    provider: Uint8Array;
+    id: MachineID;
+    cmds: Uint8Array[];
+}
+
+export interface RoflmarketQueuedCommand {
+    id: CommandID;
+    cmd: Uint8Array;
+}
+
+export interface RoflmarketProviderQuery {
+    provider: Uint8Array;
+}
+
+export interface RoflmarketOfferQuery {
+    provider: Uint8Array;
+    id: OfferID;
+}
+
+export interface RoflmarketInstanceQuery {
+    provider: Uint8Array;
+    id: MachineID;
+}
+
+export interface RoflmarketStakeThresholds {
+    provider_create: BaseUnits;
+}
+
+export interface RoflmarketParameters {}
+
+// Event types
+export interface RoflmarketProviderCreatedEvent {
+    address: Uint8Array;
+}
+
+export interface RoflmarketProviderUpdatedEvent {
+    address: Uint8Array;
+}
+
+export interface RoflmarketProviderRemovedEvent {
+    address: Uint8Array;
+}
+
+export interface RoflmarketInstanceCreatedEvent {
+    provider: Uint8Array;
+    id: MachineID;
+}
+
+export interface RoflmarketInstanceUpdatedEvent {
+    provider: Uint8Array;
+    id: MachineID;
+}
+
+export interface RoflmarketInstanceAcceptedEvent {
+    provider: Uint8Array;
+    id: MachineID;
+}
+
+export interface RoflmarketInstanceCancelledEvent {
+    provider: Uint8Array;
+    id: MachineID;
+}
+
+export interface RoflmarketInstanceRemovedEvent {
+    provider: Uint8Array;
+    id: MachineID;
+}
+
+export interface RoflmarketInstanceCommandQueuedEvent {
+    provider: Uint8Array;
+    id: MachineID;
+}
+
+// Types for rofl module
+
+export interface RoflQuotePolicy {
+    ias?: {
+        disabled?: boolean;
+        allowed_quote_statuses?: number[];
+        gid_blacklist?: number[];
+        min_tcb_evaluation_data_number?: number;
+    };
+    pcs?: {
+        disabled?: boolean;
+        tcb_validity_period: number;
+        min_tcb_evaluation_data_number: number;
+        fmspc_blacklist?: string[];
+        tdx?: {
+            // TDX-specific policy fields
```

Comment:
we also have oasis.types.NotModeled for this kind of thing

Reply:
Oh! nice!